pkg/api/customization/kontainerdriver/store.go

```diff
@@ -0,0 +1,49 @@
+package kontainerdriver
+
+import (
+	//"fmt"
+
+	"fmt"
+	errorsutil "github.com/pkg/errors"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/apis/management.cattle.io/v3/schema"
+	"github.com/rancher/types/client/management/v3"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+type Store struct {
+	types.Store
+	KontainerDriverLister v3.KontainerDriverLister
+}
+
+// Delete removes the KontainerDriver if it is not in use by a cluster
+func (s *Store) Delete(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
+	//need to get the full driver since just the id is not enough see if a cluster uses it
+	driver, err := s.KontainerDriverLister.Get("", id)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
Is this clear?

Reply:
It needs to return if the err is IsNotFound. 
```
if err != nil {
    !errors.IsNotFound(err) {
        wrap error
    }
    return obj, nil
}
````